extensions/opentelemetry/opentelemetry-exporter-otlp-http/runtime/src/main/java/io/quarkus/opentelemetry/exporter/otlp/http/runtime/LateBoundBatchSpanProcessor.java

```diff
@@ -0,0 +1,112 @@
+package io.quarkus.opentelemetry.exporter.otlp.http.runtime;
+
+import org.jboss.logging.Logger;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+
+/**
+ * Class to facilitate a delay in when the worker thread inside {@link BatchSpanProcessor}
+ * is started, enabling Quarkus to instantiate a {@link io.opentelemetry.api.trace.TracerProvider}
+ * during static initialization and set a {@link BatchSpanProcessor} delegate during runtime initialization.
+ */
+public class LateBoundBatchSpanProcessor implements SpanProcessor {
```

Comment:
I think we need to move this to the main extension, so we don't have a duplicate between this one and the one in the Jaeger extension.

Also, we are going to need to make sure that we can run both extensions together since OTel allows you to push to multiple exporters.

Reply:
> I think we need to move this to the main extension, so we don't have a duplicate between this one and the one in the Jaeger extension.

Makes sense. Even without this PR we have two identical classes with this name.

> Also, we are going to need to make sure that we can run both extensions together since OTel allows you to push to multiple exporters.

Makes sense. I would however suggest that we take care of that in a separate PR, as it isn't directly related to this feature. What do you think?

Reply:
Sure. Thanks!

Reply:
On the other hand, I've been looking a bit into this, and I'm not exactly sure how this will fit with the OTel AutoConfiguration extension (which I would like to adopt in Quarkus)